src/shared/components/Tooltip/Tooltip.tsx

```diff
@@ -1,49 +1,56 @@
 import React, { useState } from 'react'
-import { StyledTooltip, ChildrenContainer } from './Tooltip.style'
+import Tippy from '@tippyjs/react/headless'
+import { CSSTransition } from 'react-transition-group'
+import { transitions } from '@/shared/theme'
+import { Text } from '@/shared/components'
+import { StyledTooltip, IconWrapper, Arrow } from './Tooltip.style'
+import { SvgGlyphInfo } from '@/shared/icons'
 
+type Placement = 'top-start' | 'top-end' | 'bottom-start' | 'bottom-end'
 export type TooltipProps = {
   text: string
-  above?: boolean
-  right?: boolean
-  offsetY?: number
+  icon?: boolean
+  placement?: Placement
   arrowDisabled?: boolean
-  darkenContent?: boolean
   className?: string
 }
 
 const Tooltip: React.FC<TooltipProps> = ({
-  children,
   text,
-  above,
-  right,
-  offsetY,
+  icon,
+  children,
+  placement = 'bottom-start',
   arrowDisabled,
-  darkenContent = true,
   className,
 }) => {
-  const [isActive, setActive] = useState(false)
-
+  const [isVisible, setIsVisible] = useState(false)
   return (
-    <StyledTooltip
-      data-text={text}
-      isActive={isActive}
-      arrowDisabled={arrowDisabled}
-      above={above}
-      right={right}
-      offsetY={offsetY}
-      className={className}
+    <Tippy
+      onMount={() => setIsVisible(true)}
+      onHide={() => setIsVisible(false)}
+      placement={placement}
+      offset={[0, 8]}
+      render={(attrs) => (
+        <CSSTransition
+          in={isVisible}
+          timeout={parseInt(transitions.timings.sharp)}
+          classNames={transitions.names.fade}
+          unmountOnExit
+        >
+          <StyledTooltip {...attrs} className={className}>
+            {icon && (
+              <IconWrapper>
+                <SvgGlyphInfo />
+              </IconWrapper>
+            )}
+            <Text variant="caption">{text}</Text>
+            {!arrowDisabled && <Arrow />}
+          </StyledTooltip>
+        </CSSTransition>
+      )}
     >
-      <ChildrenContainer
-        onMouseEnter={() => setActive(true)}
-        onMouseLeave={() => setActive(false)}
-        onFocus={() => setActive(false)}
-        onBlur={() => setActive(false)}
-        darkenContent={darkenContent}
-        tabIndex={0}
-      >
-        {children}
-      </ChildrenContainer>
-    </StyledTooltip>
+      <span tabIndex={0}>{children}</span>
```

Comment:
Why do we need this `span` wrapper?

Reply:
@kdembler  This is a workaround for children that are not forwarding the ref. Besides that, Tippy doesn't accept `ReactNode` as a child so had to do `children as React.ReactElement` assertion as well. What do you think, should we get rid of the span (this will bring back constraint regarding ref)?

Reply:
Ah right, I see. That's an interesting question - what is more painful to work with, needing refs on all components or the wrapper that we need to style and take into account in the layout

Reply:
We can try it with the wrapper, should be much less painful now that there is only one wrapper

Reply:
Not sure if span will need additional styling, Tippy might take care of that:
[https://github.com/atomiks/tippyjs-react#component-children](https://github.com/atomiks/tippyjs-react#component-children)

Reply:
Well, it's an additional element that's getting rendered so you may need some different layouts in some places but that's fine. Let's just add `className` to the span